test/tests/functional/pbs_offline_vnodes.py

```diff
@@ -374,10 +355,151 @@ def test_fail_action_startup_hook(self):
+            momB.delete_vnode_defs()
+
+        self.create_multi_vnodes(2)
+
+        # Offline only one of the moms, the other mom and her children
+        # should still be free
+        pbs_exec = self.server.pbs_conf['PBS_EXEC']
+        pbsnodes_cmd = os.path.join(pbs_exec, 'bin', 'pbsnodes')
+        pbsnodes_offline = pbsnodes_cmd + ' -o ' + momA.shortname
+        self.du.run_cmd(self.server.hostname, cmd=pbsnodes_offline, sudo=True)
+
+        # MomA should be offline
+        self.server.expect(
+            NODE, {ATTR_NODE_state: 'offline'},
+            id=momA.shortname, max_attempts=3, interval=2)
+
+        # momB and the rest of the vnodes should be free
+        self.server.expect(NODE, {ATTR_NODE_state: 'free'},
+                           id=momB.shortname, max_attempts=3,
+                           interval=2)
+
+        self.verify_other_vnodes_state('free')
+
+    def test_pbsnodes_multi_mom_offline_online(self):
+        """
+        When all of the moms reporting a vnode are offline,
+        the vnode should also be marked offline.
+        And when pbsnodes -r is used to clear the offline from at
+        least one of the moms reporting a vnode, then that vnode
+        should also get the offline cleared.
+        Note: This test needs moms to report the same set of vnodes.
+        """
+        if len(self.moms.values()) != 2:
+            self.skipTest("Provide 2 moms while invoking test")
+
+        if self.moms.values()[0].is_cpuset_mom() is True or self.moms.values()[
+                1].is_cpuset_mom() is True:
+            self.skipTest("Skipping test on cpuset moms")
+
+        momA = ""
+        momA = self.moms.values()[0]
+        momB = ""
+        momB = self.moms.values()[1]
+
+        # The moms provided to the test may have unwanted vnodedef files.
+        if momA.has_vnode_defs():
+            momA.delete_vnode_defs()
+        if momB.has_vnode_defs():
+            momB.delete_vnode_defs()
+
+        self.create_multi_vnodes(2)
+
+        # Offline both of the moms, the vnodes reported by them
+        # will also be offlined
+        pbs_exec = self.server.pbs_conf['PBS_EXEC']
+        pbsnodes_cmd = os.path.join(pbs_exec, 'bin', 'pbsnodes')
+        pbsnodes_offline = pbsnodes_cmd + ' -o ' + momA.shortname + momB.shortname
+        self.du.run_cmd(self.server.hostname, cmd=pbsnodes_offline, sudo=True)
+
+        # MomA and MomB should be offline
+        self.server.expect(
+            NODE, {ATTR_NODE_state: 'offline'},
+            id=momA.shortname, max_attempts=3, interval=2)
+
+        self.server.expect(
+            NODE, {ATTR_NODE_state: 'offline'},
+            id=momB.shortname, max_attempts=3, interval=2)
+
+        self.verify_other_vnodes_state('offline')
+
+        # Now call pbsnodes -r to clear the offline from MomA
+        pbsnodes_clear_offline = pbsnodes_cmd + ' -r ' + momA.shortname
+        self.du.run_cmd(self.server.hostname, cmd=pbsnodes_clear_offline, sudo=True)

```

Comment:
run_cmd() works better with a list of arguments, not a string. Pass pbsnodes_clear_offline = [pbsnodes_cmd, '-r', momA.shortname]